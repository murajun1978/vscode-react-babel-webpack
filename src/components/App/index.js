import React from 'react';
import './index.css';

const increment = (state, props) => ({ counter: state.counter + 1 });
const decrement = (state, props) => ({ counter: state.counter - 1 });

class App extends React.Component {
  state = { counter: 0 };

  onIncrement = () => this.setState(increment);
  onDecrement = () => this.setState(decrement);

  render() {
    return (
      <div>
        <div>
          <label data-testid="counter">{this.state.counter}</label>
        </div>
        <button onClick={this.onIncrement}>+</button>
        <button onClick={this.onDecrement}>-</button>
      </div>
    );
  }
};

export default App;
