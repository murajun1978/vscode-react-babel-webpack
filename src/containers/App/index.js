import React from 'react';
import Counter from '../../components/Pages/Counter';

const App = () => <Counter />;

export default App;
