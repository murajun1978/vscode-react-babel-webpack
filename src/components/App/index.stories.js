import React from 'react';
import { storiesOf } from '@storybook/react';
import base from 'paths.macro';
import App from './';

const storyPath = base.replace('/src/components', '');

storiesOf(`${storyPath}`, module).add('counter', () => <App />);
