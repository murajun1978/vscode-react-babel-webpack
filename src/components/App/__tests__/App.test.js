import React from 'react';
import { render } from 'react-testing-library'
import App from '../';

test('App', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
