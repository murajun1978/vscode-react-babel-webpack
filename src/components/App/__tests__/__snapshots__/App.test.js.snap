// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<div>
  <div>
    <div>
      <label
        data-testid="counter"
      >
        0
      </label>
    </div>
    <button>
      +
    </button>
    <button>
      -
    </button>
  </div>
</div>
`;
